lib/src/svg/parser_state.dart

```diff
@@ -148,6 +151,7 @@ class _Elements {
         parserState._definitions,
         parserState.rootBounds,
         parent.style,
+        currentColor: parserState.theme?.currentColor,
```

Comment:
This doesn't seem right to me. The currentColor should be inherited at this point.

Reply:
IOW, the theme should only apply it to the root, not to children.

Reply:
(here and below)

Reply:
Thanks. I am thinking of two approaches for that: 
1. Add `currentColor` to `DrawableParent`. Populate the root SVG element `DrawableRoot` with `parserState.theme?.currentColor`, populate all the following SVG group elements `DrawableGroup` with `parent.currentColor`. In this way, this proeprty would be available at each drawable through `parent.currentColor`.
2. The same as above, but instead of adding `currentColor` to `DrawableParent`, add `currentColor` to `DrawableStyle`. In this way, this property would be available at each drawable through `parent.style`.

How does the first approach sound to you? I am not sure if the second one would make sense as `currentColor` is an inherited property of the parent rather than the style of the parent.

Reply:
Perhaps the PR landed after you started looking at this, but there's already support for currentColor to work this way without an extra property - see https://github.com/dnfield/flutter_svg/pull/588

All that's really missing is a way to supply it to the root without it defaulting to black :)

Reply:
I think I confused how the currentColor property should work. If you provide this value through a widget's property, it should only apply it to the style of the root SVG element, but not to any descendant SVG elements, right?

In the example below, the circle should have a green color rather than the value of `currentColor` provided through widget's property?
```
<svg fill="green"
 width="2em" height="1em" viewBox="0 0 200 100">
    <circle cx="50" cy="50" r="45" fill="currentColor"/>
</svg>
```

Reply:
The circle would have green, yes.

Reply:
It seems that the `svg` element needs to define a [`color`](https://www.w3.org/TR/SVG11/color.html#ColorProperty) attribute for the circle to be green.

The above code produces this result:
<img width="276" alt="CleanShot 2021-09-29 at 19 37 39@2x" src="https://user-images.githubusercontent.com/26686598/135320341-f39c40e5-7ac3-4cef-86ee-e8e02cad210b.png">

Using `color` for `svg` element (rather than `fill`) produces this result:

<img width="278" alt="CleanShot 2021-09-29 at 19 38 22@2x" src="https://user-images.githubusercontent.com/26686598/135320489-134c267d-1567-4cb1-bb03-e72a3478daa9.png">